[DOC-11265] Updated Reserved Words list

julia-cb opened the pull request:
Changes in this PR:

- Added the following to the reserved words list:
   - DEFAULT
   - LATERAL
   - REPLACE
   - VECTOR
- Moved the word ISOLATION to the correct spot on the table

Also had to move other words around to keep the table from breaking.

Jira ticket: https://issues.couchbase.com/browse/DOC-11265
Updated page: https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/reservedwords.html

julia-cb committed:
Updated Reserved Words list

julia-cb committed:
Moved ISOLATION into the correct spot

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -137,139 +137,146 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
 | RAW
-
 | REALM
 | REDUCE
 | RENAME
 | xref:n1ql-language-reference/window.adoc#nulls-treatment[RESPECT]
+
+| REPLACE
```
Swap RESPECT and REPLACE to maintain alphabetical order

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -79,49 +79,49 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
 
 | DECREMENT
 | xref:n1ql-language-reference/delete.adoc[DELETE]
+| DEFAULT
```
Swap DEFAULT and DELETE to maintain alphabetical order

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -137,139 +137,146 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
 | xref:n1ql-language-reference/createfunction.adoc[LANGUAGE]
-| LAST
 
+| LATERAL
+| LAST
```
Swap LATERAL and LAST to maintain alphabetical order

simon-dew reviewed (changes requested):
Thanks for doing this @julia-cb! Just a couple of changes in the ordering.

This PR meets the requirements of DOC-11265 as it is. I've gone through the SQL++ lexer and I think I've identified all the remaining reserved words that we need to add, as per comments on the ticket. We can add those to this PR or it in a later PR if you prefer?

simon-dew commented:
As noted on the JIRA issue, most of the remaining reserved words relate to sequences, so we could add those as part of DOC-11421.

julia-cb committed:
Update list with words from Simon's comment on jira ticket (also removed SEMI)

julia-cb commented:
@simon-dew Thanks for the review Simon! I added the words you mentioned in your jira ticket comment, and removed SEMI. I'll wait to hear back from Donald on whether there's more words we need to add to the list :)

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -193,83 +207,97 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
 | REVOKE
 | RIGHT
 | ROLE
+| ROLES
```
`ROLES` shouldn't be here, sorry. Please delete. My bad, can't read my own writing.

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -137,34 +144,41 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
+| MAXVALUE
 | xref:n1ql-language-reference/merge.adoc[MERGE]
-
 | MINUS
```
I know it was in the original file, but I don't think `MINUS` shouldn't be here either. Please delete.

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -193,83 +207,97 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
 | xref:n1ql-language-reference/window.adoc#window-frame-extent[PRECEDING]
 | PREPARE
+| PREV
+| PREVIEW
```
Sorry, this should be `PREVIOUS`, not <s>PREVIEW</s>. Again, my bad, can't read my own writing.

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/reservedwords.adoc:
```diff
@@ -193,83 +207,97 @@ The following keywords are reserved and cannot be used as unescaped identifiers:
 | xref:n1ql-language-reference/window.adoc#nulls-treatment[RESPECT]
+| RESTART
+| RESTRICT
+| READ
+| RECURSIVE
```
Please move `READ` between RAW and REALM, and move `RECURSIVE` between REALM and REDUCE.

simon-dew reviewed (changes requested):
Thanks for adding the rest of the restricted words @julia-cb ! I double-checked the list against [n1ql.nex](https://github.com/couchbase/query/blob/trinity/parser/n1ql/n1ql.nex) and I have some more comments. Some of them are my fault, I obviously can't read my own writing.

julia-cb committed:
Updated page with Simon's feedback

julia-cb commented:
@simon-dew Thanks for the review again! I addressed all the changes you requested. Let me know if the PR is good to go now :)

simon-dew reviewed (approved):
All comments updated, LGTM 👍
